doc/releases/release-notes-3.6.rst

```diff
@@ -147,6 +167,11 @@ Build system and infrastructure
 
 * Fixed an issue whereby using some shields with sysbuild would cause a cmake Kconfig error.
 
+* Fixed an issue where the macros ``_POSIX_C_SOURCE`` and ``_XOPEN_SOURCE`` would be defined
+  globally when building with Picolibc or for the native (``ARCH_POSIX``) targets.
+  After this change users may need to define them for their own applications or libraries if they
+  require them.
```

Comment:
Is this worth an entry in the migration guide? I have a downstream app whose CI just broke so I guess it might happen to others, and I could definitely use some guidance there myself :) 

```
...
/Users/kartben/artificial-nose-zephyr-ws/zephyr/drivers/wifi/esp32/src/esp_wifi_drv.c: In function 'scan_done_handler':
/Users/kartben/artificial-nose-zephyr-ws/zephyr/drivers/wifi/esp32/src/esp_wifi_drv.c:204:40: error: implicit declaration of function 'strnlen'; did you mean 'strlen'? [-Werror=implicit-function-declaration]
  204 |                         int ssid_len = strnlen(ap_list_buffer[k].ssid, WIFI_SSID_MAX_LEN);
      |                                        ^~~~~~~
      |                                        strlen
cc1: all warnings being treated as errors
...
```

Reply:
Oh, that's a bug (and it got in in a PR of mine :( ) fix posted here https://github.com/zephyrproject-rtos/zephyr/pull/68233 
Anyhow, it does make sense to mention this in the migration guide, so I will.